src/membership_mesh.rs

```diff
@@ -0,0 +1,125 @@
+//! Rapid mesh-membership based functionality
+use crate::settings::Settings;
+use blip::{Member, Mesh, MeshService, Subscription};
+use chrono::Utc;
+use evmap::{ReadHandle, WriteHandle};
+use log::info;
+use rand::Rng;
+use sled;
+use std::{
+    net::{IpAddr, Ipv6Addr, SocketAddr},
+    thread,
+};
+use tonic::{
+    self,
+    transport::{self, Channel},
+};
+
+/// Attempt to join a mesh if one is available, and if not, starts one up
+/// Returns an EVMap that readers can clone and used to read active members
+pub async fn boostrap(
+    cfg: &Settings,
+) -> anyhow::Result<(String, ReadHandle<MeshKey, Vec<PeerNode>>)> {
```

Comment:
The EnvMap crate is a super cool bit of tech: allows a thread-safe, lock-free hash map like structure that's performant for our use case (single writer), multiple readers. Means that as membership list changes, downstream functionality in threads can quickly "adjust configuration" by reading the values out of the hash map (which will have been updated when we call refresh) without either passing around channels, or having to fully re-up threads/etc.